File: sample/src/main/java/ru/surfstudio/standard/interactor/common/response/ErrorResponse.java
package ru.surfstudio.standard.interactor.common.response;

import com.google.gson.annotations.SerializedName;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * базовый интерфейс ответа
 */

@Data
@AllArgsConstructor
@NoArgsConstructor
public class ErrorResponse implements BaseResponse {

    @SerializedName("userMessage")
    private String userMessage;
    @SerializedName("errorCode")
    private int errorCode;
}

File: template/src/main/java/ru/surfstudio/standard/interactor/common/response/ErrorResponse.kt
package ru.surfstudio.standard.interactor.common.response

import com.google.gson.annotations.SerializedName

/**
 * базовый интерфейс ответа
 */

open class ErrorResponse: BaseResponse {

    @SerializedName("userMessage")
    var userMessage: String? = null
        set(userMessage) {
            field = this.userMessage
        }
    @SerializedName("errorCode")
    var errorCode: Int = 0
        set(errorCode) {
            field = this.errorCode
        }
}
